src/library_syscall.js

```diff
@@ -608,6 +610,7 @@ var SyscallsLibrary = {
 #if ASSERTIONS
         assert(!res.errno);
 #endif
+        {{{ makeSetValue('addrlen', 0, 'res.salen', 'i32') }}};
```

Comment:
Given that all four of the call sites do this, eight not pass this the `res.salen` as an extra argument to __write_sockaddr?

Then it can just return errno rather than a JS object.

Also, since they are JS-only functions these should probably be called
`$writeSockaddr` and `$readSockaddr` to avoid conflict or confusion with native symbols (which start with `_`).   But we can do that a followup if you prefer.

Reply:
I changed it to extra argument and returning errno.
I'm sure if I want to rename it, I don't even know how this ends up being called `__write_sockaddr` while declared `_write_sockaddr`...

Reply:
sgtm